py-polars/polars/_typing.py

```diff
@@ -108,6 +108,7 @@ def __arrow_c_stream__(self, requested_schema: object | None = None) -> object:
 IpcCompression: TypeAlias = Literal["uncompressed", "lz4", "zstd"]
 JoinValidation: TypeAlias = Literal["m:m", "m:1", "1:m", "1:1"]
 Label: TypeAlias = Literal["left", "right", "datapoint"]
+MaintainOrder: TypeAlias = Literal["none", "left", "right", "left_right", "right_left"]
```

Comment:
Can we use python `None`  directly (so not part of this set.

The argument then becomes `arg: MaintainOrder | None`

PS. I also think we should name it `MaintainOrderJoin`